src/Type/Definition/ResolveInfo.php

```diff
@@ -266,4 +388,61 @@ private function foldSelectionSet(SelectionSetNode $selectionSet, int $descend):
 
         return $fields;
     }
+
+    /**
+     * @throws InvariantViolation|Error|\Exception
+     *
+     * @return array<string>
+     */
+    private function foldSelectionWithAlias(SelectionSetNode $selectionSet, int $descend, Type $parentType): array
+    {
+        /** @var array<string, bool> $fields */
+        $fields = [];
+
+        foreach ($selectionSet->selections as $selectionNode) {
+            if ($selectionNode instanceof FieldNode) {
+                $fieldName = $selectionNode->name->value;
+                $aliasName = $selectionNode->alias->value ?? $fieldName;
+
+                if ($fieldName === Introspection::TYPE_NAME_FIELD_NAME) {
+                    continue;
+                }
+
+                assert($parentType instanceof HasFieldsType, 'ensured by query validation and the check above which excludes union types');
```

Comment:
Is this copy-pasted from somewhere? The comment makes little sense to me in this context.

Reply:
I was using the same logical as QueryPlan, we don't want stuff like __typename and we need to have fieldsType.
But I can remove this if you want.